### ISSUE TITLE
Generators: reconsider -f --samples defaults

### ISSUE THREAD

**Issue by thorehusfeldt** [RagnarGrootKoerkamp/BAPCtools#304]
The generators framework work very smoothly, but I now constantly get tripped up in the following dance:

```bash
> bt generate
PROBLEM closestpairscoring
Generate: sample/2                    SKIPPED: 2.ans; supply -f --samples to overwrite
Generate: sample/1                    SKIPPED: 1.ans; supply -f --samples to overwrite
Generate: sample/3                    SKIPPED: 3.ans; supply -f --samples to overwrite
^[[A^CFATAL ERROR: Running interrupted[######################---------------------------------------------------------------------------------- 11/48]
> bt generate -f --samples
PROBLEM closestpairscoring
Generate: sample/2                    CHANGED 2.ans
Generate: sample/1                    CHANGED 1.ans
Generate: sample/3                    CHANGED 3.ans
```
The `-f --samples` tradition targets a workflow where sample typically are provided _as files_ in `data/samples`, but the new framework much prefers to have everything in `generators/generators.yaml`.

This confuses two recommendations for what “the expected workflow” is; an author that follows best practice should be supported by default settings.

Suggestions:
1. drop the default `SKIPPED` behaviour
2. introduce `config.I_know_what_Im_doing_shut_up` or `config.this_contest_has_some_people_with_dirty_samples` somewhere (presumably with a better name)
3. make BAPC check if _all files in `data` are mentioned in `generators.yaml`_. If so, assume the author knows what they’re doing and set the default to `-f --sample`. (This is very hard to document an can lead to unexpected behaviour.)

**Comment by RagnarGrootKoerkamp** [comment]
1. I think dropping the `--samples` part of `-f --samples` is OK and we can simply warn when this happens. If you care about specific sample output you can hardcode/copy them. (Otherwise there is also usually `git` to recover what you lost.)
2. I'm not a fan of introducing such flags. Good default behavior for power users is more convenient.
3. See #305. Indeed I think we should (always/by default) check that all files in `data/` are generated, to simplify towards a single workflow of generating everything.

**Comment by mzuenni** [comment]
Current idea for `bt generate`:

1. remove all files in `tmp/problemdir/removed` that are older than ~10min
2. identify all files that are up to date
3. move all files that are not up to date to `tmp/problemdir/removed` (and print a warning with the path to this dir if this happens)
4. generate all missing files

- This should resolve the `-f --sample` behavior by just always regenerating them
- This should resolve #305 c by always (re)moving non generated files (with the mentioned warning)
- We don't do anything irrevocably (directly) since you can recover stuff from the tmpdir
- This hopefully simplifies code by removing stuff like `move unlisted` and having unlisted testcases in the first place

**Comment by RagnarGrootKoerkamp** [comment]
Fixed in #348. Just `bt generate` will now overwrite everything and move old/unwanted files to `/tmp` so they can be recovered if needed.
